### PR TITLE
Task completed: GP-124. Ammo clip + reload

### PR DESCRIPTION
**Pull request by frnzee**
- added and implemented ammo clip class
- fixed input actions map for reloading
- added reload prefab
- added reload event

### CODE REVIEW

File: Assets/Scripts/Gameplay/Weapons/AmmoClip.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            {
+                return _bulletAmount;
+            }
+            set
+            {
+                _bulletAmount = Mathf.Clamp(value, 0, _maxBulletAmount);
+            }
+        }
+
+        public bool Shot()
```
Comment (ega-ega) [inline Assets/Scripts/Gameplay/Weapons/AmmoClip.cs]: Renam to waste bullet

File: Assets/Scripts/Gameplay/Weapons/AmmoClip.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
+using System;
+using UnityEngine;
+
+namespace Gameplay.Weapons
+{
+    public class AmmoClip : IWeapon
```
Comment (ega-ega) [inline Assets/Scripts/Gameplay/Weapons/AmmoClip.cs]: AmmoClip dont need implemnted IWeapon

File: Assets/Scripts/Gameplay/Weapons/AmmoClip.cs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    {
+        private int _bulletAmount;
```
Comment (ega-ega) [inline Assets/Scripts/Gameplay/Weapons/AmmoClip.cs]: rename to _amount

File: Assets/Scripts/Gameplay/Weapons/AmmoClip.cs [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        private int _maxBulletAmount;
```
Comment (ega-ega) [inline Assets/Scripts/Gameplay/Weapons/AmmoClip.cs]: rename to _size

**Review by ega-ega — CHANGES_REQUESTED** [review]

File: Assets/Scripts/Gameplay/Weapons/AmmoClip.cs (hunk 1 above)
Comment (frnzee) [reply Assets/Scripts/Gameplay/Weapons/AmmoClip.cs]: done

File: Assets/Scripts/Gameplay/Weapons/AmmoClip.cs (hunk 2 above)
Comment (frnzee) [reply Assets/Scripts/Gameplay/Weapons/AmmoClip.cs]: done

File: Assets/Scripts/Gameplay/Weapons/AmmoClip.cs (hunk 3 above)
Comment (frnzee) [reply Assets/Scripts/Gameplay/Weapons/AmmoClip.cs]: done

File: Assets/Scripts/Gameplay/Weapons/AmmoClip.cs (hunk 4 above)
Comment (frnzee) [reply Assets/Scripts/Gameplay/Weapons/AmmoClip.cs]: done

File: Assets/Scripts/Gameplay/Weapons/AmmoClip.cs [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        
+            return false;
+        }
+
+        public void Reload()
+        {
+            _amount = _size;
+        }
+
+        public bool HasAmmo()
```
Comment (ega-ega) [inline Assets/Scripts/Gameplay/Weapons/AmmoClip.cs]: change to property

File: Assets/Scripts/Gameplay/Character/ICanReload.cs [hunk 6]
```diff
@@ -0,0 +1,11 @@
+using System;
+
+namespace Gameplay
+{
+    public interface ICanReload
+    {
+        float ReloadTime { get; }
```
Comment (ega-ega) [inline Assets/Scripts/Gameplay/Character/ICanReload.cs]: remove from the interface

File: Assets/Scripts/Gameplay/Character/ICanReload.cs [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+        void Reload();
+        event Action ReloadChanged;
```
Comment (ega-ega) [inline Assets/Scripts/Gameplay/Character/ICanReload.cs]: rename to Reloaded

File: Assets/Scripts/Gameplay/Weapons/IWeapon.cs [hunk 8]
```diff
@@ -2,6 +2,8 @@ namespace Gameplay.Weapons
 {
     public interface IWeapon : IWeaponReadonly
     {
-        bool Shot();
+        bool WasteBullet();
```
Comment (ega-ega) [inline Assets/Scripts/Gameplay/Weapons/IWeapon.cs]: Rename to Shot() please restore

File: Assets/Scripts/Gameplay/Character/Character.cs [hunk 9]
```diff
@@ -26,6 +26,8 @@ public class Character : MonoBehaviour, ICharacter, IDamageable, IHaveHealth, IC
         public float MoveSpeed => _stats.MoveSpeed;
         
         public float AttackSpeed => _stats.AttackSpeed;
+        public float ReloadTime { get; }
```
Comment (ega-ega) [inline Assets/Scripts/Gameplay/Character/Character.cs]: Pls remove

File: Assets/Scripts/Gameplay/Weapons/Weapon.cs [hunk 10]
```diff
@@ -53,13 +56,14 @@ private void Update()
 
         public bool Shot()
         {
-            if (_shotCooldownTimer > 0 || _bulletAmount == 0)
+            if (_shotCooldownTimer > 0 || !_ammoClip.HasAmmo)
             {
+                Debug.Log("Need reload");
```
Comment (ega-ega) [inline Assets/Scripts/Gameplay/Weapons/Weapon.cs]: Pls Remove

**Review by ega-ega — APPROVED** [review]

File: Assets/Scripts/Gameplay/Weapons/AmmoClip.cs (hunk 5 above)
Comment (frnzee) [reply Assets/Scripts/Gameplay/Weapons/AmmoClip.cs]: done

File: Assets/Scripts/Gameplay/Character/ICanReload.cs (hunk 6 above)
Comment (frnzee) [reply Assets/Scripts/Gameplay/Character/ICanReload.cs]: done

File: Assets/Scripts/Gameplay/Character/ICanReload.cs (hunk 7 above)
Comment (frnzee) [reply Assets/Scripts/Gameplay/Character/ICanReload.cs]: done

File: Assets/Scripts/Gameplay/Weapons/IWeapon.cs (hunk 8 above)
Comment (frnzee) [reply Assets/Scripts/Gameplay/Weapons/IWeapon.cs]: done

File: Assets/Scripts/Gameplay/Character/Character.cs (hunk 9 above)
Comment (frnzee) [reply Assets/Scripts/Gameplay/Character/Character.cs]: done

**Review by ega-ega — APPROVED** [review]

**Review by ega-ega — APPROVED** [review]

File: Assets/Scripts/Gameplay/Weapons/Weapon.cs (hunk 10 above)
Comment (frnzee) [reply Assets/Scripts/Gameplay/Weapons/Weapon.cs]: done

**Pull request merged**